bids/reports/report.py

```diff
@@ -0,0 +1,115 @@
+"""Generate publication-quality data acquisition methods section from BIDS dataset.
+"""
+from __future__ import print_function
+import json
+from os.path import dirname
+from os.path import abspath
+from os.path import join as pathjoin
+from collections import Counter
+
+from bids.reports import utils
+
+
+class BIDSReport(object):
+    """
+    Generates publication-quality data acquisition methods section from BIDS
+    dataset.
+
+    Parameters
+    ----------
+    layout : :obj:`bids.grabbids.BIDSLayout`
+        Layout object for a BIDS dataset.
+    config : :obj:`str` or :obj:`dict`, optional
+        Configuration info for methods generation. Can be a path to a file
+        (str), a dictionary, or None. If None, loads and uses default
+        configuration information.
+    """
+    def __init__(self, layout, config=None):
+        self.layout = layout
+        if config is None:
+            root = dirname(abspath(__file__))
+            config = pathjoin(root, 'config', 'converters.json')
+            with open(config) as fobj:
+                config = json.load(fobj)
+
+        if not isinstance(config, dict):
+            raise ValueError('Input config must be None, dict, or path to json '
+                             'file containing dict.')
+
+        self.config = config
+
+    def generate(self, task_converter=None):
+        """Generate the methods section.
+
+        Parameters
+        ----------
+        task_converter : :obj:`dict`
+            A dictionary with information for converting task names from BIDS
+            filename format to human readable strings.
+        """
+        if task_converter is None:
+            task_converter = {'rest': 'resting state'}
+
+        self.config['task'] = task_converter
+
+        descriptions = []
+
+        subjs = self.layout.get_subjects()
+        for sid in subjs:
+            description = self._report(subj=sid)
+            descriptions.append(description)
+        counter = Counter(descriptions)
+        print('Number of patterns detected: {0}'.format(len(counter.keys())))
+        print(utils.warnings())
```

Comment:
This is a bit pedantic, but maybe call this `reminder` or something. If it's at the level of a legit warning, it should probably be issued via `warnings.warn` rather than printed.

Reply:
Yeah.. that wasn't the best naming on my part. Reminder makes more sense.